tests/manage/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py

```diff
@@ -0,0 +1,372 @@
+                        bg_wrap.wrap,
+                        cluster_exp_helpers.cluster_copy_ops,
+                        p,
+                        iterations=200)
+                )
+
+        # Start pvc ops in the background.:
+        logger.info('Started pvc create delete operations')
+        executor_run_bg_ios_ops.submit(
+            bg_wrap.wrap,
+            test_create_delete_pvcs,
+            multi_pvc_factory,
+            pod_factory,
+            project,
+            iterations=200
+        )
+
+        # Start NooBaa IOs in the background.:
+        logger.info("Started s3_io_create_delete...")
+
+        executor_run_bg_ios_ops.submit(
+            bg_wrap.wrap,
+            s3_io_create_delete,
+            mcg_obj,
+            awscli_pod,
+            bucket_factory,
+            iterations=200)
+
+        logger.info("Started obc_io_create_delete...")
+
+        executor_run_bg_ios_ops.submit(
+            bg_wrap.wrap,
+            obc_io_create_delete,
+            mcg_obj,
+            awscli_pod,
+            bucket_factory,
+            iterations=200)
+
+        # All ocs nodes are in Ready state (including master):
+        executor_run_bg_ios_ops.submit(
+            bg_wrap.wrap, check_nodes_status, iterations=100)
+
+        # Get restart count of ocs pods before expanstion
+        restart_count_before = pod_helpers.get_pod_restarts_count(
+            defaults.ROOK_CLUSTER_NAMESPACE)
+
+        # Get osd pods before expansion
+        osd_pods_before = pod_helpers.get_osd_pods()
+
+        # Get the total space in cluster before expansion
+        ct_pod = pod_helpers.get_ceph_tools_pod()
+        output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph osd df')
+        total_space_b4_expansion = int(output.get('summary').get('total_kb'))
+        logger.info(f"total_space_b4_expansion == {total_space_b4_expansion}")
+
+        logger.info("############## Calling add_capacity $$$$$$$$$$")
+
+        #####################
+        # Call add_capacity #
+        #####################
+        osd_size = storage_cluster.get_osd_size()
+        result = storage_cluster.add_capacity(osd_size)
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+
+        # New osd (all) pods corresponding to the additional capacity should be
+        # in running state
+        pod.wait_for_resource(
+            timeout=1200,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=result * 3
+        )
+
+        #################################
+        # Exit criteria verification:   #
+        #################################
+        BackgroundOps.EXPANSION_COMPLETED = True
+
+        # No ocs pods should get restarted unexpectedly
+        # Get restart count of ocs pods after expansion and see any pods got
+        # restated
+        restart_count_after = pod_helpers.get_pod_restarts_count(
```

Comment:
Is it possible to export some of the checks here in 'Exit criteria verification' to another file? then we can use it easily in other tests too. For example we can write a function that checks the right number of OSDs are added: 
```def check_pods_after_expansion(osds_pods_before):
    osd_pods_after = pod_helpers.get_osd_pods()
    number_of_osds_added = len(osd_pods_after) - len(osd_pods_before)
    logger.info(f"### number_of_osds_added = {number_of_osds_added}, "
                f"before = {len(osd_pods_before)}, after = {len(osd_pods_after) }")
    # If the difference b/w updated count of osds and old osd count is not
    # 3 then expansion failed
    return number_of_osds_added == 3
```

Reply:
I'm making this test as tier2, will address this comment in my next pr